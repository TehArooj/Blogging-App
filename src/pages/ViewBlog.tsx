import React, { useEffect, useState } from "react";
import { BiArrowBack } from "react-icons/bi";
import { useNavigate } from "react-router-dom";
import { auth } from "../utils/firebase/firebase.utils";
import { ArrowIcon } from "../assests/Arrow";

function ViewBlog() {
  const navigate = useNavigate();

  const [username, setUsername] = useState("");

  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (user) {
        setUsername(user.displayName!);
        localStorage.setItem("userName", user.displayName!);
      }
      if (!localStorage.getItem("userName")) {
        navigate("/login");
      }
    });
  }, [navigate, username]);
  const back = () => {
    navigate(-1);
  };
  return (
    <>
      <div className="grid grid-cols-12">
        <div className="col-span-2 m:col-span-1 z-50">
          <div className="mt-10">
            <div className=" ml-20 top-11 tb:ml-12 ">
              <div onClick={back}>
                <span className="text-xl tb:text-base">Back</span>
              </div>
            </div>
            <div className="absolute top-12 ml-14 tb:ml-8">
              <div onClick={back}>
                <BiArrowBack className="text-4xl text-secondary  tb:text-2xl  " />
                {/* <ArrowIcon />*/}
              </div>
            </div>
          </div>
        </div>
        <div className="col-span-8 flex flex-col mt-10  md:mb-20 tb:mb-16 m:mb-14  m:ml-11">
          <div className="flex flex-col items-left  m:mt-10">
            <h1 className="text-4xl text-justify text-primary mb-5 font-dm font-normal m:text-3xl m:mt-10">
              15 Disadvantages Of Freedom And How You Can Workaround It.
            </h1>
            <div className="flex justify-between">
              <div className="text-secondary mb-10 text-xl font-extralight  m:text-base">
                written by @samurai992 <br />
                on 26 August 2022
              </div>
            </div>
            <p className="text-xl font-normal text-justify first-letter:text-7xl  m:text-base m:font-normal">
              Et molestiae hic earum repellat aliquid est doloribus delectus.
              Enim illum odio porro ut omnis dolor debitis natus. Voluptas
              possimus deserunt sit delectus est saepe nihil. Qui voluptate
              possimus et quia. Eligendi voluptas voluptas dolor cum. Rerum est
              quos quos id ut molestiae fugit. Et molestiae hic earum repellat
              aliquid est doloribus delectus. Enim illum odio porro ut omnis
              dolor debitis natus. Voluptas possimus deserunt sit delectus est
              saepe nihil. Qui voluptate possimus et quia. Eligendi voluptas
              voluptas dolor cum. Rerum est quos quos id ut molestiae fugit. Et
              molestiae hic earum repellat aliquid est doloribus delectus. Enim
              illum odio porro ut Tehreems. Voluptas possimus deserunt sit
              delectus est saepe nihil. Qui voluptate possimus et quia. Eligendi
              voluptas voluptas dolor cum. Rerum est quos quos id ut molestiae
              fugit.
            </p>
          </div>
        </div>
        <div className="col-span-2"></div>
      </div>
    </>
  );
}

export default ViewBlog;
