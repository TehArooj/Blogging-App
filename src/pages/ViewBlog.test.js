import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { cleanup } from "@testing-library/react";
import ViewBlog from "./ViewBlog";

afterEach(cleanup);

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
      addEventListener: function () {},
    };
  };


it("<ViewBlog /> renders correctly", async () => {
  const tree = renderer
    .create(
      <Router>
        <ViewBlog />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<ViewBlog /> check matchers and pass", async () => {
  const blog = {
    date: new Date(),
    blog:'Blog Text',
    title: 'LeBron James new book stars',
    author:'Maxwell'
  };

  expect(blog).toMatchSnapshot({
    date: expect.any(Date),
    blog: expect.any(String),
    title: expect.any(String),
    author: expect.any(String),
  });
});
