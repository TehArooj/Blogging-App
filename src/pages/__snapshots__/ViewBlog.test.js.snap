// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewBlog /> check matchers and pass 1`] = `
Object {
  "author": Any<String>,
  "blog": Any<String>,
  "date": Any<Date>,
  "title": Any<String>,
}
`;

exports[`<ViewBlog /> renders correctly 1`] = `
<div
  className="grid grid-cols-12"
>
  <div
    className="col-span-2 tb:col-span-1 tb:z-50 m:col-span-1 m:z-50"
  >
    <div
      className="mt-11"
    >
      <div
        className=" ml-16 top-9 font-dm font-normal m:ml-12"
      >
        <div
          onClick={[Function]}
        >
          <span
            className="text-2xl cursor-pointer"
          >
            Back
          </span>
        </div>
      </div>
      <div
        className="absolute top-16 ml-11 m:ml-7"
      >
        <div
          className="cursor-pointer"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="23"
            viewBox="0 0 70 23"
            width="70"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0.939331 10.3766C0.353546 10.9624 0.353546 11.9121 0.939331 12.4979L10.4853 22.0439C11.0711 22.6296 12.0208 22.6296 12.6066 22.0439C13.1924 21.4581 13.1924 20.5083 12.6066 19.9225L4.12132 11.4373L12.6066 2.95197C13.1924 2.36619 13.1924 1.41644 12.6066 0.830654C12.0208 0.244867 11.0711 0.244867 10.4853 0.830654L0.939331 10.3766ZM69.0669 9.93726L1.99999 9.93726V12.9373L69.0669 12.9373V9.93726Z"
              fill="#A5A5A5"
              opacity="0.4"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="absolute top-[45%] left-[45%] m:top-[40%] m:left-[40%]"
  >
    <div
      aria-busy={true}
      aria-label="ball-triangle-loading"
      data-testid="ball-triangle-loading"
      role="status"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <svg
        data-testid="ball-triangle-svg"
        height={100}
        stroke="#4fa94d"
        viewBox="0 0 57 57"
        width={100}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="5"
              cy="50"
              r={5}
            >
              <animate
                attributeName="cy"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                repeatCount="indefinite"
                values="50;5;50;50"
              />
              <animate
                attributeName="cx"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                repeatCount="indefinite"
                values="5;27;49;5"
              />
            </circle>
            <circle
              cx="27"
              cy="5"
              r={5}
            >
              <animate
                attributeName="cy"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                from="5"
                repeatCount="indefinite"
                to="5"
                values="5;50;50;5"
              />
              <animate
                attributeName="cx"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                from="27"
                repeatCount="indefinite"
                to="27"
                values="27;49;5;27"
              />
            </circle>
            <circle
              cx="49"
              cy="50"
              r={5}
            >
              <animate
                attributeName="cy"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                repeatCount="indefinite"
                values="50;50;5;50"
              />
              <animate
                attributeName="cx"
                begin="0s"
                calcMode="linear"
                dur="2.2s"
                from="49"
                repeatCount="indefinite"
                to="49"
                values="49;5;27;49"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <div
    className="col-span-2"
  />
</div>
`;
